RecoBTag/DeepFlavour/interface/deep_helpers.h

```diff
@@ -0,0 +1,71 @@
+#ifndef RecoBTag_DeepFlavour_deep_helpers_h
+#define RecoBTag_DeepFlavour_deep_helpers_h
+
+#include "DataFormats/GeometryCommonDetAlgo/interface/Measurement1D.h"
+#include "DataFormats/BTauReco/interface/TaggingVariable.h"
+
+#include "TrackingTools/IPTools/interface/IPTools.h"
+
+#include "DataFormats/BTauReco/interface/CandIPTagInfo.h"
+
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/Candidate/interface/VertexCompositePtrCandidate.h"
+//#include "RecoVertex/VertexTools/interface/VertexDistanceXY.h"
+
+namespace btagbtvdeep {
+
+  // remove infs and NaNs with value  (adapted from DeepNTuples)
+  const float catch_infs(const float in,
+				const float replace_value);
```

Comment:
I think this split to .h and .cc was done to avoid some issue with circular dependencies when they were inlined... @cvernier @kpedro88 may be able to comment more

Reply:
It's been about 6 months since I looked at this. It might have been an ODR violation.

Reply:
After some more thought/recollection, I think the problem was this:
* if there is just a .h file and no .cc file, it doesn't become its own translation unit
* if the .h file is then included in multiple other files, it appears in multiple translation units -> ODR violated
* with a .h and .cc file together, it becomes its own translation unit
* that translation unit is then linked to others -> no ODR issue

If *every* function were inlined, splitting to .h and .cc might be avoided. But I'm not sure if this is reasonable (not all functions were inlined in the previous version of this file).